audio_service/android/src/main/java/com/ryanheise/audioservice/AudioServicePlugin.java

```diff
@@ -766,17 +770,35 @@ public void onMethodCall(MethodCall call, Result result) {
             Map<?, ?> args = (Map<?, ?>)call.arguments;
             switch (call.method) {
             case "setMediaItem": {
-                Map<?, ?> rawMediaItem = (Map<?, ?>)args.get("mediaItem");
-                MediaMetadataCompat mediaMetadata = createMediaMetadata(rawMediaItem);
-                AudioService.instance.setMetadata(mediaMetadata);
-                result.success(null);
+                Handler handler = new Handler(Looper.getMainLooper());
+                Executors.newSingleThreadExecutor().execute(() -> {
+                    try {
+                        Map<?, ?> rawMediaItem = (Map<?, ?>)args.get("mediaItem");
+                        MediaMetadataCompat mediaMetadata = createMediaMetadata(rawMediaItem);
+                        AudioService.instance.setMetadata(mediaMetadata);
+                        handler.post(() -> result.success(null));
+                    } catch (Exception e) {
+                        handler.post(() -> {
+                            result.error("UNEXPECTED_ERROR", "Unexpected error", Log.getStackTraceString(e));
+                        });
+                    }
+                });
                 break;
             }
             case "setQueue": {
-                @SuppressWarnings("unchecked") List<Map<?, ?>> rawQueue = (List<Map<?, ?>>)args.get("queue");
-                List<MediaSessionCompat.QueueItem> queue = raw2queue(rawQueue);
-                AudioService.instance.setQueue(queue);
-                result.success(null);
+                Handler handler = new Handler(Looper.getMainLooper());
+                Executors.newSingleThreadExecutor().execute(() -> {
+                     try {
```

Comment:
This block is indented 5 spaces instead of 4.

Reply:
fixed